src/Microsoft.Azure.SignalR.Management/RestHubLifetimeManager.cs

```diff
@@ -92,31 +123,119 @@ public override Task SendUsersAsync(IReadOnlyList<string> userIds, string method
+        {
+            switch (statusCode)
+            {
+                case HttpStatusCode.BadRequest:
+                    {
+                        throw new AzureSignalRInvalidArgumentException(requestUri, innerException, detail);
+                    }
+                case HttpStatusCode.Unauthorized:
+                    {
+                        throw new AzureSignalRUnauthorizedException(requestUri, innerException);
+                    }
+                case HttpStatusCode.NotFound:
+                    {
+                        throw new AzureSignalRInaccessibleEndpointException(requestUri, innerException);
+                    }
+                default:
+                    {
+                        throw new AzureSignalRRuntimeException(requestUri, innerException);
+                    }
+            }
+        }
+
+        private static void ThrowExceptionOnNetworkConnectivityAndOtherIssues(Exception innerException, string requestUri, string detail = null)
+        {
+            switch (innerException)
+            {
+                case HttpRequestException hrex:
+                    {
+                        throw new AzureSignalRInaccessibleEndpointException(requestUri, innerException); // network connectivity issue
+                    }
+                default:
+                    {
+                        throw new AzureSignalRRuntimeException(requestUri, innerException); // other issues
+                    }
+            }
+        }
+
+        private static HttpRequestMessage BuildRequest(RestApiEndpoint endpoint, HttpMethod httpMethod, string methodName = null, object[] args = null, CancellationToken cancellationToken = default)
+        {
+            var payload = httpMethod == HttpMethod.Post ? new PayloadMessage { Target = methodName, Arguments = args } : null;
+            return GenerateHttpRequest(endpoint.Audience, payload, endpoint.Token, HttpMethod.Post);
+        }
+
+        private static async Task CallRestApiAsync(HttpRequestMessage request)
+        {
+            var httpClient = HttpClientFactory.CreateClient();
+            HttpResponseMessage response = null;
+            string detail = "";
+
+            try
+            {
+                response = await httpClient.SendAsync(request);
+            }
+            catch (HttpRequestException ex)
+            {
+                ThrowExceptionOnNetworkConnectivityAndOtherIssues(ex, request.RequestUri.ToString()); // not known host
+            }
+
+            try
+            {
+                detail = await response.Content.ReadAsStringAsync();
+                response.EnsureSuccessStatusCode();
+            }
+            catch (HttpRequestException ex)
+            {
+                ThrowExceptionOnResponseFailure(ex, response.StatusCode, request.RequestUri.ToString(), detail);
+            }
+            catch (Exception ex)
```

Comment:
Do not catch Exception

Reply:
`ReadAsStringAsync` might throw `LimitMemoryStream`, `HttpRequestException`, `ArgumentOutOfRangeException`, `InvalidOperationException` ...
Except for `HttpRequestException`, other exceptions are only related to customer's machine. Thus I think catch `HttpRequestException` is enough. What do you think? @chenkennt

Reply:
Then you don't need to catch those exceptions.